src/mineral.ts

```diff
@@ -8,13 +8,17 @@ interface Mineral extends RoomObject {
     readonly prototype: Mineral;
 
     /**
+     * Room cannot be undefined for a Mineral.
+     */
+    room: Room;
+    /**
      * The remaining amount of resources.
      */
     mineralAmount: number;
     /**
      * The resource type, one of the RESOURCE_* constants.
      */
-    mineralType: string;
+    mineralType: ResourceConst;
```

Comment:
Wouldn't the API be rather more fluent if `ResourceConst` was instead called `ResourceType`?

Doesn't seem necessary to explicitly call a string literal type "const". Of course it's constant: it's a set of literals :P